### DISCUSSION TITLE
Type only import?

### DISCUSSION THREAD

**Discussion by httpete** [dotansimha/graphql-code-generator#9103] [Q&A] [answered]
I have auto generated types, but the result has:

import { TypedDocumentNode as DocumentNode } from '@graphql-typed-document-node/core';

Which throws up in storybook, because it should be flagged as a type only import:

import type { TypedDocumentNode as DocumentNode } from '@graphql-typed-document-node/core';

Works. Is there a way to tell it to use that syntax? It is really more correct.

**Comment by httpete** [comment, marked as answer]
I found the answer! For anyone else:

```
  generates: {
    'gql/': {
      preset: 'client-preset',
      config: {
        useTypeImports: true,
      },
    },
  },

**Reply by productdevbook** [reply]
How. to export type ? this only imports.
